### PR TITLE
Persist user defined test emails as user meta

### LINKED ISSUE #156: Prepopulate Test Email With User Account Email
A nice enhancement would be to have the "Send A Test To" field come pre-populated with the email associated with the current WordPress user.

### PR DESCRIPTION
**Pull request by miguelpeixe**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Populate the field for email testing with the address(es) previously set by the user. If no email has been previously set, populates the field with the user email address.

Closes #156.

### How to test the changes in this Pull Request:

1. Write a newsletter
2. Send a test to a custom email address (or multiple comma-separated addresses)
3. Refresh the page or create a new newsletter and see that the emails have persisted

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Review by dkoo — CHANGES_REQUESTED** [review]
Works as described! The only issue I saw is that if you update the email addresses in the input field, then collapse the Testing sidebar panel, the value will revert to the original value if you re-expand the Testing panel (even after sending a test email). We'll need to figure out a way to update the state in that case.

The easiest way I can think of is to move the state value from the `Testing` component up to the `NewsletterEdit` component and then pass it as a prop to `Testing` so that the state will persist even if `Testing` is unmounted.

**Comment by miguelpeixe** [comment]
I've also changed a little bit of the field behavior. Now the default email is the user email address. If the user sets a new email, it is saved as user meta.

File: src/newsletter-editor/testing/index.js [hunk 1]
```diff
@@ -26,48 +26,48 @@ export default compose( [
 			savePost,
 		};
 	} ),
-] )( ( { apiFetchWithErrorHandling, inFlight, postId, savePost, setInFlightForAsync } ) => {
-	const [ testEmail, setTestEmail ] = useState(
-		window?.newspack_newsletters_data?.user_test_emails?.join( ',' ) || ''
-	);
+] )(
+	( { apiFetchWithErrorHandling, inFlight, postId, savePost, setInFlightForAsync, ...props } ) => {
+		const { testEmail, onChangeEmail } = props;
```
Comment (dkoo) [inline src/newsletter-editor/testing/index.js:31]: Non-blocking nitpick: why not destructure these variables on the previous line along with the other props?

**Review by dkoo — APPROVED** [review]
Works well, and I like how the field now defaults to the user's email! (Which is what #156 originally suggested, so now we have the best of both worlds.) I have a non-blocking question below, but feel free to merge if you did it for a specific reason.

File: src/newsletter-editor/testing/index.js (hunk 1 above)
Comment (miguelpeixe) [reply src/newsletter-editor/testing/index.js:31]: You're right! I wanted to have them together for legibility but doesn't look like we have a problem with having them together with the other component props.

**Review by dkoo — APPROVED** [review]

File: src/newsletter-editor/testing/index.js (hunk 1 above)
Comment (dkoo) [reply src/newsletter-editor/testing/index.js]: Got it, looks good now!

**Pull request merged**